dll/win32/advapi32/reg/reg.c

```diff
@@ -3492,8 +3492,14 @@ RegOpenUserClassesRoot(IN HANDLE hToken,
                                      NULL,
                                      0,
                                      &RequiredLength);
-    if (!NT_SUCCESS(Status) && (Status != STATUS_BUFFER_TOO_SMALL))
+    if (Status != STATUS_BUFFER_TOO_SMALL)
     {
+        if (Status == STATUS_SUCCESS)
```

Comment:
There are about 2 billion other success status codes.
I'd suggest
```
if (NT_SUCCESS(Status))
{
    ...
}
else if (Status != STATUS_BUFFER_TOO_SMALL)
{
    ...
}
```

Reply:
> `NT_SUCCESS(Status)`

Obviously. Fixed.

> `if else if`

While that is somewhat clearer to read, I feel it is better to have only one block of "error" handling, especially as we always expect `STATUS_BUFFER_TOO_SMALL` result.